plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/XtextAntlrGeneratorFragment2.xtend

```diff
@@ -0,0 +1,176 @@
+		file.content = '''
+			public class �grammar.parserClass.simpleName� extends �AbstractAntlrParser� {
+			
+				@�Inject�
+				private �grammar.grammarAccess� grammarAccess;
+
+				@Override
+				protected void setInitialHiddenTokens(�XtextTokenStream� tokenStream) {
+					tokenStream.setInitialHiddenTokens(�FOR hidden : grammar.initialHiddenTokens SEPARATOR ", "�"�hidden�"�ENDFOR�);
+				}
+			
+				@Override
+				protected �grammar.internalParserClass� createParser(�XtextTokenStream� stream) {
+					return new �grammar.internalParserClass�(stream, getGrammarAccess());
+				}
+			
+				@Override 
+				protected String getDefaultRuleName() {
+					return "�grammar.allParserRules.head.originalElement.name�";
+				}
+			
+				public �grammar.grammarAccess� getGrammarAccess() {
+					return this.grammarAccess;
+				}
+			
+				public void setGrammarAccess(�grammar.grammarAccess� grammarAccess) {
+					this.grammarAccess = grammarAccess;
+				}
+			}
+		'''
+		file
+	}
+	
+	def JavaFileAccess generateAntlrTokenFileProvider() {
+		val file = fileFactory.createGeneratedJavaFile(grammar.getAntlrTokenFileProviderClass)
+		file.content = '''
+			public class �grammar.antlrTokenFileProviderClass.simpleName� implements �IAntlrTokenFileProvider� {
+			
+				@Override
+				public �InputStream� getAntlrTokenFile() {
+					�ClassLoader� classLoader = getClass().getClassLoader();
+					return classLoader.getResourceAsStream("�grammar.tokenFileName�");
+				}
+			}
+		'''
+		file
+	}
+	
+	override checkConfiguration(Issues issues) {
+		super.checkConfiguration(issues)
+		if (getOptions().isBacktrackLexer()) {
+			issues.addError("This fragment does not support the option 'backtracking' for the lexer. Use 'org.eclipse.xtext.xtext.generator.parser.antlr.ex.rt.AntlrGeneratorFragment2' instead");
+		}
+		if (getOptions().isIgnoreCase()) {
+			issues.addError("This fragment does not support the option 'ignorecase'. Use 'org.eclipse.xtext.xtext.generator.parser.antlr.ex.rt.AntlrGeneratorFragment2' instead");
+		}
+	}
+	
+	def addBindingsAndImports() {
+		if (projectConfig.runtimeManifest !== null) {
+			projectConfig.runtimeManifest=>[
+				exportedPackages += #[
+					grammar.parserPackage,
+					grammar.internalParserPackage
+				]
+				requiredBundles += "org.antlr.runtime"
+			]
+		}
+		val rtBindings = new GuiceModuleAccess.BindingFactory()
+			.addTypeToType(IParser.typeRef, grammar.parserClass)
+			.addTypeToType(ITokenToStringConverter.typeRef,AntlrTokenToStringConverter.typeRef)
+			.addTypeToType(IAntlrTokenFileProvider.typeRef, grammar.antlrTokenFileProviderClass)
+			.addTypeToType(Lexer.typeRef, grammar.lexerClass)
+			.addTypeToType(ITokenDefProvider.typeRef, AntlrTokenDefProvider.typeRef)
+			.addTypeToProviderInstance(grammar.lexerClass, '''�LexerProvider�.create(�grammar.lexerClass�.class)''')
+			.addConfiguredBinding("RuntimeLexer", '''
+					binder.bind(�Lexer�.class)
+					.annotatedWith(�Names�.named(�LexerBindings�.RUNTIME))
```

Comment:
Cosmetics: please increase indentation in multiline expressions, e.g on line 2 and 3 for the `binder....class); `